src/openms/include/OpenMS/KERNEL/MSExperiment.h

```diff
@@ -1278,6 +1278,9 @@ std::vector<MSChromatogram> extractXICs(
     /// Are all MSSpectra in this experiment part of an IM Frame? I.e. they all have the same RT, but different drift times
     bool isIMFrame() const;
 
+	// Check if there are any MS2 spectra with ion mobility data
```

Comment:
@cbielow there is also spec.hasIMData()
any comment on how to query the presence of scan level drift time IM and binary IM data arrays?

Reply:
check out https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/IONMOBILITY/IMTypes.h#L71
Exists for MSExperiment and MSSpectrum.

Not sure what happens when MSExperiment is not fully loaded but IM is in floatdataarrays though.
The mzMLHandler will know...
However, I guess you are interested in spectra with a single drift time, i.e. IMFormat::MULTIPLE_SPECTRA only.
Or are there search engines which can analyse spectra with IM annotated in floatdataarrays (IMFormat::CONCATENATED)?

So this:
```
if (IMTypes::determineIMFormat(exp) == IMFormat::MULTIPLE_SPECTRA)
{
  ... go for annotation...
}
```
should do.

Reply:
you can also repeat this check for each spectrum if you anticipate MSExperiments with IM + non-IM spectra, e.g.
```
const auto im_format = IMTypes::determineIMFormat(exp); // this is "expensive"
if (im_format == IMFormat::MULTIPLE_SPECTRA || im_format == IMFormat::MIXED)
{
  for (auto spec : ... spectra ...)
  {
    if (IMTypes::determineIMFormat(spec) == IMFormat::MULTIPLE_SPECTRA)
    { // has valid '.getDriftTime()'
    }
  }

}

Reply:
Ok thanks for the input. I think it makes more sense to put this logic in `SpectrumMetaDataLookup::addMissingIMToPeptideIDs`

I dont anticipate an MSExperiment containing MS2 spectra with both, IM and non IM information

Reply:
IMFormat::MULTIPLE_SPECTRA is the one I am looking for